apps/server/src/shared/domain/service/file-copy-append.service.ts

```diff
@@ -56,39 +56,20 @@ export class FileCopyAppendService {
 	}
 
 	private createFailedCopyStatus(taskCopyStatus: CopyStatus) {
-		let fileGroupStatus = this.getFileGroupStatus(taskCopyStatus.elements);
-		if (fileGroupStatus === undefined) {
-			fileGroupStatus = {
-				type: CopyElementType.FILE_GROUP,
-				status: CopyStatusEnum.FAIL,
-			};
-		}
-		const elements =
-			fileGroupStatus.elements && fileGroupStatus.elements.length > 0
-				? fileGroupStatus.elements.map((el) => {
-						el.status = CopyStatusEnum.FAIL;
-						return el;
-				  })
-				: undefined;
+		const fileGroupStatus = {
+			type: CopyElementType.FILE_GROUP,
+			status: CopyStatusEnum.FAIL,
+		};
 		const updatedFileGroupStatus = {
 			...fileGroupStatus,
 			status: CopyStatusEnum.FAIL,
-			elements,
 		};
-		taskCopyStatus.status = this.copyHelperService.deriveStatusFromElements(taskCopyStatus.elements as CopyStatus[]);
 		taskCopyStatus.elements = this.setFileGroupStatus(taskCopyStatus.elements, updatedFileGroupStatus);
+		taskCopyStatus.status = this.copyHelperService.deriveStatusFromElements(taskCopyStatus.elements);
 		return taskCopyStatus;
 	}
 
-	private getFileGroupStatus(elements: CopyStatus[] = []): CopyStatus | undefined {
-		return elements.find((el) => el.type === CopyElementType.FILE_GROUP) as CopyStatus;
-	}
-
 	private setFileGroupStatus(elements: CopyStatus[] = [], updatedFileGroupStatus: CopyStatus): CopyStatus[] {
```

Comment:
If the replace code is not needed anymore and the fileGroupStatus is always only appended, we can remove setFileGroupStatus and increase readability by moving the [...elements, updatedFileGroupStauts] to the places where the function was used.